src/setfit/modeling.py

```diff
@@ -683,6 +683,18 @@ def sentence_pairs_generation_cos_sim(sentences, pairs, cos_sim_matrix):
     return pairs
 
 
+def sentence_pairs_remove_duplicates(sentences: List[InputExample]):
+    key_pairs = set()
+    rm_duplicate_sentences = []
+    for s in sentences:
+        key = tuple(sorted(s.texts))
+        if key not in key_pairs:
+            key_pairs.add(key)
+            if key[0] != key[1]:
+                rm_duplicate_sentences.append(s)
+    return rm_duplicate_sentences
```

Comment:
This differs slightly from my understanding of #258. In particular, this function:
1. removes duplicate pairs
2. removes pairs where the items are identical

I believe that #258 only mentions the former. Did you find the pairs where both items being identical to be a problem?

Reply:
No. Tbh I wrote these changes ages ago but only got round to evaluating on public dataset/ PR yesterday. 
For my use I was slashing as much data as possible to speed up training, this is a remnant from that I forgot I added !

I can run some tests again with 2. applied or not, and we can see what the analysis shows. My intuition would say labelling identical pairs wouldn't improve performance, but we can see..

But will hold off until have some resolution about it should be applied (based on your first comment @tomaarsen) 👍

Reply:
I agree it seems unintuitive to generate a list then reduce it. My original thinking was conceptually it seems understandable to say "oh we've removed 80% of the data as duplicates, that will greater speed up my training time". On your idea of generating the "proper" list I've also added more fundamental thoughts on #258.

I'd be happy to add the `unique_pairs` logic, it's sensible and in fact I think it could really just go in as a default rather than optional parameter - as when would you want to randomly duplicate pairs over incorporating new data?

The only outstanding point for me is in the case of >2 classes, `remove_duplicate_samples` would start removing more positive samples, as they are sampled evenly with negative pairs, despite being far fewer. In these cases the runtime speed ups come mostly from removing excessive positive pairs, which the tests have shown (at high sample numbers atleast) to have little impact on the accuracy.

Again happy to add `unique_pairs` as PR, but can put it in a separate PR maybe for now?
Let me know if have any thoughts on this, thanks @tomaarsen.